### ISSUE TITLE
[Game/Heroic Issue] GTA V

### ISSUE THREAD

**Issue by allora** [CommandMC/EpicLinux#7]
### Game Name

GTA V

### What is happening?

After closing GTA V, heroic throws an error that causes heroic to crash to desktop: 
```
(00:37:58) INFO:    [Legendary]:       Launching Grand Theft Auto V: LD_PRELOAD= WINEPREFIX=/home/<user>/Games/Heroic/Prefixes/GTAV WINEESYNC=1 WINEFSYNC=1 /usr/bin/mangohud --dlsym /usr/bin/gamemoderun /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 9d2d0eb64d5c44529cece33fe2a46482 --language en --wine /home/<user>/.config/heroic/tools/wine/Wine-GE-Proton7-32/bin/wine --wine-prefix /home/<user>/Games/Heroic/Prefixes/GTAV
(00:43:08) ERROR:   [Backend]:         RangeError: Invalid string length
```
Repro steps:
- Open GTA V
- Quit

Additionally after this, the game logs are blank. Also note: GTA V has a Social Club error when closing, which may cause this invalid string length sine the game doesnt end as expected. Sadly since the game logs are busted.. I dont have much to add.

### Game log

https://pastebin.com/VYu7VzWc

### System information

System:
  Host: rufus Kernel: 6.0.5-arch1-1 arch: x86_64 bits: 64 Desktop: bspwm
    v: 0.9.10 Distro: Arch Linux
Memory:
  RAM: total: 62.74 GiB used: 14.21 GiB (22.6%)
  RAM Report: permissions: Unable to run dmidecode. Root privileges
    required.
CPU:
  Info: 16-core model: AMD Ryzen 9 5950X bits: 64 type: MT MCP cache:
    L2: 8 MiB
  Speed (MHz): avg: 2449 min/max: 2200/5083 cores: 1: 2200 2: 2200 3: 2200
    4: 2200 5: 2200 6: 2200 7: 2200 8: 2200 9: 2200 10: 2200 11: 3400 12: 2200
    13: 3197 14: 2200 15: 2200 16: 2200 17: 2879 18: 2878 19: 3400 20: 2200
    21: 2200 22: 2879 23: 2200 24: 3400 25: 2200 26: 2200 27: 2200 28: 2200
    29: 2871 30: 2200 31: 2878 32: 2200
Graphics:
  Device-1: AMD Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M]
    driver: amdgpu v: kernel
  Display: server: X.Org v: 21.1.4 with: Xwayland v: 22.1.4 driver: X:
    loaded: amdgpu dri: radeonsi gpu: amdgpu resolution: 1: 1920x1080 60Hz
    2: 640x480 60Hz 3: 2560x1440 60Hz
  OpenGL: renderer: AMD Radeon RX 6700 XT (navi22 LLVM 14.0.6 DRM 3.48
    6.0.5-arch1-1) v: 4.6 Mesa 22.2.1

### Wine Version

Wine-GE 7.32

### Common issues

- [X] I have verified that this issue is not resolved by installing DXVK/vkd3d-proton
- [X] I have verified that this issue is not resolved by enabling virtual desktop
- [X] I have verified that this issue is also occurring in a clean Wineprefix (meaning a prefix that has only this one game installed)

**Comment by StefanLobbenmeier** [comment]
did you follow https://github.com/CommandMC/EpicLinux/wiki/GTA-V ?

**Comment by allora** [comment]
Yes. To be clear, GTA V works fine. This happens when exiting the game (or in the case of the benchmark, when the game resets), heroic crashes. The game itself runs fine. It exists fine. Heroic just crashes when the game is closed.

Beta 2.5 actually doesnt crash heroic, which is nice, but still pops up the error..

**Comment by CommandMC** [comment]
This is definitely an issue exclusive to Heroic (so this is not really the right place for it)

At some point, I've made Heroic write output to log files directly (instead of keeping it in a buffer and writing it to the log file at the end). I'm not sure if this made it into the 2.4.3 release
You might be seeing the effects of that here
